libraries/AP_GPS/AP_GPS_UAVCAN.cpp

```diff
@@ -91,9 +100,11 @@ AP_GPS_Backend* AP_GPS_UAVCAN::probe(AP_GPS &_gps, AP_GPS::GPS_State &_state)
             } else {
                 _detected_modules[i].driver = backend;
                 backend->_detected_module = i;
-                debug_gps_uavcan(2,
-                                 _detected_modules[i].ap_uavcan->get_driver_index(),
-                                 "Registered UAVCAN GPS Node %d on Bus %d\n",
+                if (_state.devid != _detected_modules[i].node_id) {
+                    _state.devid = _detected_modules[i].node_id;
+                    _gps.set_and_save_devid(_state.instance, _detected_modules[i].node_id);
```

Comment:
Doesn't this leave you with (if 142 is the one that was actually still around):

```
GPS_DEVID 142
GPS2_DEVID 142
```

This is bad because the user is now confused which one is around/connected, as well as invalidating their POS parameters.